Build fails with Boost 1.87.0 due to use of removed `io_service`

### Bug Description:

Trying to build Manticoresearch (both 7.0.0 and master) with Boost 1.87.0 fails with: 
```
src/searchdbuddy.cpp:17:10: fatal error: 'boost/asio/io_service.hpp' file not found
   17 | #include <boost/asio/io_service.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is due to usage of `io_service` which was deprecated back in 1.66.0 (https://github.com/boostorg/asio/commit/b60e92b13ef68dfbb9af180d76eae41d22e19356) and removed in 1.87.0 (https://github.com/boostorg/asio/commit/ec0908c562102915423d8bd7aefd3079efbb6c86)

---

The fix is a simple replacement (e.g. `s/io_service/io_context/g`) as there is already a minimum requirement of Boost 1.71.0 which guarantees `io_context` exists:
https://github.com/manticoresoftware/manticoresearch/blob/13c0380dff6da04c3629be3a926930d13c97d161/src/CMakeLists.txt#L138-L139

### Manticore Search Version:

7.0.0; master (13c0380)

### Operating System Version:

macOS 15

### Have you tried the latest development version?

Yes

### Internal Checklist:

<sup>To be completed by the assignee. Check off tasks that have been completed or are not applicable.</sup>
<details>

- [ ] Implementation completed
- [ ] Tests developed
- [ ] Documentation updated
- [ ] Documentation reviewed
- [ ] [Changelog](https://docs.google.com/spreadsheets/d/1mz_3dRWKs86FjRF7EIZUziUDK_2Hvhd97G0pLpxo05s/edit?pli=1&gid=1102439133) updated

</details>

--- a/src/searchdbuddy.cpp
+++ b/src/searchdbuddy.cpp
@@ -14,7 +14,7 @@
 #include "netreceive_ql.h"
 #include "client_session.h"
 
-#include <boost/asio/io_service.hpp>
+#include <boost/asio/io_context.hpp>
 #include <boost/asio/read_until.hpp>
 #include <boost/process.hpp>
 #if _WIN32
@@ -32,7 +32,7 @@ static CSphString g_sUrlBuddy;
 static CSphString g_sStartArgs;
 
 static const int PIPE_BUF_SIZE = 2048;
-static std::unique_ptr<boost::asio::io_service> g_pIOS;
+static std::unique_ptr<boost::asio::io_context> g_pIOS;
 static std::vector<char> g_dPipeBuf ( PIPE_BUF_SIZE );
 static CSphVector<char> g_dLogBuf ( PIPE_BUF_SIZE );
 static std::unique_ptr<boost::process::async_pipe> g_pPipe;
@@ -357,7 +357,7 @@ BuddyState_e TryToStart ( const char * sArgs, CSphString & sError )
 	g_pPipe.reset();
 	g_pIOS.reset();
 
-	g_pIOS.reset ( new boost::asio::io_service );
+	g_pIOS.reset ( new boost::asio::io_context );
 	g_pPipe.reset ( new boost::process::async_pipe ( *g_pIOS ) );
 
 	std::unique_ptr<boost::process::child> pBuddy;

